packages/components/src/RBACProvider/RBACProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Access } from '@redhat-cloud-services/rbac-client';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+
+import {
+  RBACContext,
+  doesHavePermissions,
+  getRBAC,
+  hasAllPermissions,
+  initialPermissions,
+} from '@redhat-cloud-services/frontend-components-utilities/RBAC';
+
+const hasAccessWithUserPermissions = (userPermissions: (Access | string)[]) => {
+  return (requiredPermissions: string[], checkAll = true): boolean => {
+    return checkAll ? hasAllPermissions(userPermissions, requiredPermissions) : doesHavePermissions(userPermissions, requiredPermissions);
+  };
+};
+
+export interface RBACProviderProps {
+  appName: string;
+}
+
+export const RBACProvider: React.FunctionComponent<RBACProviderProps> = ({ appName, children }) => {
```

Comment:
Can we also have optional loader component? Which would defaults to
```
<Bullseye>
  <Spinner size="xl" />
</Bullseye>
```

Reply:
Will add that, yes!